tamr_toolbox/utils/custom_button.py

```diff
@@ -0,0 +1,422 @@
+        ---
+
+    Returns:
+        Path to yaml file created
+    """
+    # URL & pagename validation
+    invalid_urls = []
+    invalid_pages = []
+    for button_dict in buttons:
+        if button_dict["buttonType"] == "postButton":
+            url = button_dict["postUrl"]
+        else:
+            url = button_dict["redirectUrl"]
+
+        if not url.startswith(("http://", "https://")):
+            invalid_urls.append(url)
+
+        invalid_pages += [
+            p for p in button_dict["pageNames"] if not _check_valid_page_name(pagename=p)
+        ]
+
+    if len(invalid_urls) > 0 and len(invalid_pages) > 0:
+        value_error_message = f"Invalid url and pagenames. \
+            url(s) {invalid_urls} must begin with http:// or https:// \
+            invalid page name(s): {invalid_pages}"
+        LOGGER.error(value_error_message)
+        raise ValueError(value_error_message)
+    elif len(invalid_urls) > 0 and len(invalid_pages) == 0:
+        value_error_message = f"Invalid url(s) {invalid_urls}. Must begin with http:// or https://"
+        LOGGER.error(value_error_message)
+        raise ValueError(value_error_message)
+    elif len(invalid_urls) == 0 and len(invalid_pages) > 0:
+        value_error_message = (
+            f"Invalid pagename(s): {invalid_pages}. See docs for allowed Tamr button page names"
+        )
+        LOGGER.error(value_error_message)
+        raise ValueError(value_error_message)
+
+    file = f"{extension_name}.yaml"
+    filepath = os.path.join(output_dir, file)
+
+    output_dict = {"extensionName": extension_name, "buttons": buttons}
+
+    LOGGER.info(f"Saving {file} to {output_dir}")
+    with open(f"{filepath}", "w") as yaml_file:
+        yaml.dump(output_dict, yaml_file, sort_keys=False)
+
+    return filepath
+
+
+def register_buttons(
+    *,
+    buttons: List[str],
+    tamr_install_dir: str,
+    remote_client: Optional["paramiko.SSHClient"] = None,
+    impersonation_username: Optional[str] = None,
+    impersonation_password: Optional[str] = None,
+):
+    """Registers a list of buttons in a Tamr instance. Requires Tamr restart to display buttons in UI.
+
+    Runs in a remote environment if an ssh client is specified otherwise runs in the local shell.
+    If an impersonation_username is provided, the command is run as the provided user.
+    If an impersonation_password is provided, password authentication is used for impersonation,
+    otherwise sudo is used.
+
+    Args:
+        buttons: A list of yaml files (absolute paths) with button configs
+        tamr_install_dir: Full path to directory where Tamr is installed
+        remote_client: An ssh client providing a remote connection
+        impersonation_username: A bash user to run the command as,
+            this should be the tamr install user
+        impersonation_password: The password for the impersonation_username
+
+    Returns:
+    """
+    LOGGER.info(f"Registering the following buttons in Tamr: {buttons}")
+
```

Comment:
These functions will only work when ran as scripts **directly on the server**. Which is the most common way I have seen pipelines implemented at customers. In my opinion it would be too complicated to create and get the yaml files onto the servers remotely.